projects/kit/components/input-phone-international/input-phone-international.component.ts

```diff
@@ -0,0 +1,246 @@
+    protected readonly textfieldOptions = inject(TuiTextfieldOptionsDirective, {
+        optional: true,
+    });
+
+    protected open = false;
+    protected textfieldValue = '';
+
+    @ViewChild(MaskitoDirective, {read: ElementRef})
+    private readonly inputPhone?: ElementRef<HTMLInputElement>;
+
+    @ViewChild(TuiSelectDirective, {read: ElementRef})
+    private readonly countrySelect?: ElementRef<HTMLSelectElement>;
+
+    @Input()
+    public countries = this.options.countries;
+
+    @Output()
+    public readonly countryIsoCodeChange = this.countryIsoCode$.pipe(
+        distinctUntilChanged(),
+        skip(1),
+    );
+
+    @Input('countryIsoCode')
+    public set isoCode(code: TuiCountryIsoCode) {
+        this.countryIsoCode$.next(code);
+    }
+
+    public get nativeFocusableElement(): HTMLElement | null {
+        return this.inputPhone && !this.computedDisabled
+            ? this.inputPhone.nativeElement
+            : null;
+    }
+
+    public get focused(): boolean {
+        return Boolean(
+            tuiIsNativeFocused(this.countrySelect?.nativeElement) ||
+                tuiIsNativeFocused(this.inputPhone?.nativeElement) ||
+                this.open,
+        );
+    }
+
+    public onPaste(event: Event, phonesMetadata: MetadataJson | null): void {
+        if (
+            !tuiIsInputEvent(event) ||
+            !phonesMetadata ||
+            (!event.inputType.includes('Drop') && !event.inputType.includes('Paste'))
+        ) {
+            return;
+        }
+
+        const newValue = event.data || '';
+        const prefixedValue = newValue.startsWith(CHAR_PLUS)
+            ? newValue
+            : CHAR_PLUS + newValue;
+
+        if (validatePhoneNumberLength(prefixedValue) === 'TOO_SHORT') {
+            return;
+        }
+
+        const countryIsoCode = maskitoGetCountryFromNumber(prefixedValue, phonesMetadata);
+
+        if (countryIsoCode) {
+            this.countryIsoCode$.next(countryIsoCode);
+        }
+    }
+
+    public onItemClick(isoCode: TuiCountryIsoCode): void {
+        this.open = false;
+        this.countryIsoCode$.next(isoCode);
+        this.nativeFocusableElement?.focus();
+    }
+
+    constructor() {
+        super();
+        combineLatest([this.countryIsoCode$, this.options.metadata])
+            .pipe(
+                map(args => this.computeMask(...args)),
+                takeUntilDestroyed(),
+            )
+            .subscribe(this.mask$);
+    }
+
+    public override setDisabledState(): void {
+        super.setDisabledState();
+    }
+
+    public override writeValue(unmaskedValue: string): void {
+        super.writeValue(unmaskedValue);
+
+        const maskOptions = this.mask$.value;
+
+        this.textfieldValue = maskOptions
+            ? maskitoTransform(unmaskedValue, maskOptions)
+            : unmaskedValue; // it will be calibrated later when mask is ready (by maskitoInitialCalibrationPlugin)
+        this.cdr.detectChanges();
```

Comment:
Do we really need it and not markForCheck?

Reply:
Does `formControl.patchValue` trigger tick of application ? 🤔

Reply:
I've try to replace it by `markForCheck`.
This test fails:

https://github.com/taiga-family/taiga-ui/blob/c8a24a8d70ade7674f83222f1aa8239af1055a6f/projects/demo-cypress/src/tests/input-phone-international.cy.ts#L227-L231

Reply:
Ok then, let's keep it like this for now